### PR TITLE
Copy: Refactor variant report code and remove old functions

### PR DESCRIPTION
**Pull request by matren395**
Copy of 'Refactor variant report code and remove old functions' (link: https://github.com/broadinstitute/gnomad_methods/pull/575) with only changes to relevant files to clear up some very odd problems with Git History.

### CODE REVIEW

File: gnomad/utils/annotations.py [hunk 1]
```diff
@@ -2391,7 +2154,7 @@ def _create_group_dicts(
                 sex_list = []
                 for sex in ["XX", "XY"]:
                     sex_key = f"{group}-{sex}-adj"
-                    sex_index_value = ht.freq_index_dict.get(sex_key)
+                    sex_index_value = frequency_index.get(sex_key)
                     sex_label = f"{group}.{sex}"
```
Comment (klaricch) [inline gnomad/utils/annotations.py:2158]: "sex_label" -> "sex_id"

File: gnomad/utils/annotations.py [hunk 2]
```diff
@@ -2406,61 +2169,47 @@ def _create_group_dicts(
             list_of_group_info_dicts.append(group_result)
 
     # Overall frequency, via label 'adj' which is currently stored at
```
Comment (klaricch) [inline gnomad/utils/annotations.py:2171]:
```suggestion
    # Add overall frequency, via label 'adj' which is currently stored at
```

File: gnomad/utils/annotations.py [hunk 3]
```diff
@@ -2406,61 +2169,47 @@ def _create_group_dicts(
...
-                "version": f"{label_version}",
-            },
-            "focusAllele": "",  # TODO load from vrs_json table
-            "focusAlleleCount": overall_freq["AC"],
-            "locusAlleleCount": overall_freq["AN"],
-            "alleleFrequency": overall_freq["AF"],
-            "cohort": {"id": "ALL"},
-        }
-    )
+    # Final dictionary to be returned.
```
Comment (klaricch) [inline gnomad/utils/annotations.py:2175]:
```suggestion
    # Create final dictionary to be returned.
```

File: gnomad/utils/annotations.py [hunk 4]
```diff
@@ -2406,61 +2169,47 @@ def _create_group_dicts(
...
+    ancillaryResults = {"homozygotes": overall_freq["homozygote_count"]}
+
+    if input_struct.popmax:
+        ancillaryResults["popMaxFAF95"] = {
+            "frequency": input_struct.popmax.faf95,
+            "confidenceInterval": 0.95,
+            "popFreqId": f"{gnomad_id}.{input_struct.popmax.pop.upper()}",
+        }
 
     # Read coverage statistics if a table is provided
```
Comment (klaricch) [inline gnomad/utils/annotations.py:2202]:
```suggestion
    # Add mean coverage statistics if a coverage Table is provided
```

File: gnomad/utils/annotations.py [hunk 5]
```diff
@@ -2406,61 +2169,47 @@ def _create_group_dicts(
...
+        "id": f"{label_name}-{label_version}-{gnomad_id}",
+        "type": "CohortAlleleFrequency",
+        "label": f"Overall Cohort Allele Frequency for {gnomad_id}",
+        "derivedFrom": {
+            "id": f"{label_name}{label_version}",
+            "type": "DataSet",
+            "label": f"{label_name} v{label_version}",
+            "version": f"{label_version}",
+        },
+        "focusAllele": "",  # TODO load from VRS dictionary
```
Comment (klaricch) [inline gnomad/utils/annotations.py:2186]: remaining TODO?

**Review by klaricch — CHANGES_REQUESTED** [review]

File: gnomad/utils/annotations.py [hunk 6]
```diff
@@ -2244,73 +1994,88 @@ def gks_compute_seqloc_digest_batch(
     return ht.drop("vrs_json").join(ht_with_location_parsed, how="left")
 
 
-def add_gks_vrs(ht: hl.Table):
+def add_gks_vrs(
+    input_locus: hl.struct,
```
[the remark this answers is not in the document]
Comment (matren395) [reply gnomad/utils/annotations.py:1998]: this is right actually , thank you

Comment (matren395) [inline gnomad/utils/annotations.py:1998]:
```suggestion
    input_locus: hl.locus,
```

File: gnomad/utils/annotations.py [hunk 7]
```diff
@@ -2244,73 +1994,88 @@ def gks_compute_seqloc_digest_batch(
...
-    label_name: str,
-    label_version: str,
+    input_struct: hl.struct,
+    label_name: str = "gnomAD",
+    label_version: str = "3.1.2",
     coverage_ht: hl.Table = None,
     ancestry_groups: list = None,
     ancestry_groups_dict: dict = None,
     by_sex: bool = False,
+    frequency_index: dict = None,
```
Comment (matren395) [inline gnomad/utils/annotations.py:2052]:
```suggestion
    freq_index_dict: dict = None,
```

File: gnomad/utils/annotations.py (hunk 1 above)
Comment (matren395) [inline gnomad/utils/annotations.py:2158]:
```suggestion
                    sex_id = f"{group}.{sex}"
```

File: gnomad/utils/annotations.py (hunk 5 above)
Comment (matren395) [reply gnomad/utils/annotations.py:2186]: ah, the TODO means that it is filled in later inside of `gnomad.py` , but then again that probably isn't the best usage maybe? Would there be a better way to say "this is empty now but it is filled in later, dont worry" or does that even need said

File: gnomad/utils/annotations.py [hunk 8]
```diff
@@ -2244,73 +1994,88 @@ def gks_compute_seqloc_digest_batch(
...
+        If None, omit coverage in return.
+    :param ancestry_groups: List of strings of shortened names of cohorts to return results for.
+        Example: ['afr','fin','nfe'] . Default is None.
+    :param ancestry_groups_dict: Dict mapping shortened genetic ancestry group names to full names.
+        Example: {'afr':'African/African American'} . Default is None.
+    :param by_sex: Boolean to include breakdown of cohorts by inferred sex (XX and XY) as well.
+        Default is None.
+    :param vrs_only: Boolean to return only VRS information and no general frequency information.
+        Default is False.
+    :frequency_index: Dict mapping groups to their index for freq info in ht.freq_index_dict[0].
```
Comment (matren395) [inline gnomad/utils/annotations.py:2075]:
```suggestion
    :freq_index_dict: Dict mapping groups to their index for freq info in ht.freq_index_dict[0].
```

File: gnomad/utils/annotations.py [hunk 9]
```diff
@@ -2391,7 +2154,7 @@ def _create_group_dicts(
                 sex_list = []
                 for sex in ["XX", "XY"]:
                     sex_key = f"{group}-{sex}-adj"
-                    sex_index_value = ht.freq_index_dict.get(sex_key)
+                    sex_index_value = frequency_index.get(sex_key)
```
Comment (matren395) [inline gnomad/utils/annotations.py:2157]:
```suggestion
                    sex_index_value = freq_index_dict.get(sex_key)
```

File: gnomad/utils/annotations.py [hunk 10]
```diff
@@ -2375,11 +2138,11 @@ def _create_group_dicts(
     # different ancestry groups to.
     list_of_group_info_dicts = []
 
-    # Iterate through provided groups and generate dictionaries
+    # Iterate through provided groups and generate dictionaries.
     if ancestry_groups:
         for group in ancestry_groups:
             key = f"{group}-adj"
-            index_value = ht.freq_index_dict.get(key)
+            index_value = frequency_index.get(key)
```
Comment (matren395) [inline gnomad/utils/annotations.py:2145]:
```suggestion
            index_value = freq_index_dict.get(key)
```

File: gnomad/resources/grch38/gnomad.py [hunk 11]
```diff
@@ -585,57 +514,47 @@ def gnomad_gks_batch(
...
-            va_freq_dict = json.loads(ann.gks_va_freq_json)  # Hail Struct as json
+            va_freq_dict = add_gks_va(
+                input_struct=variant,
+                label_name="gnomAD",
+                label_version=version,
+                coverage_ht=coverage_ht,
+                ancestry_groups=pops_list,
+                ancestry_groups_dict=POP_NAMES,
+                by_sex=by_sex,
+                frequency_index=ht.freq_index_dict.collect()[0],
```
Comment (matren395) [inline gnomad/resources/grch38/gnomad.py:550]:
```suggestion
                freq_index_dict=ht.freq_index_dict.collect()[0],
```

File: gnomad/utils/annotations.py (hunk 8 above)
Comment (matren395) [reply gnomad/utils/annotations.py:2075]: this change overlaps with another, will commit later

Comment (matren395) [reply gnomad/utils/annotations.py:2073]: incorporated in a commit from VSCode

File: gnomad/utils/annotations.py (hunk 5 above)
Comment (klaricch) [reply gnomad/utils/annotations.py:2184]: TODO is usually used in python scripts to indicate something that still needs to be worked on (https://medium.com/imdoneio/the-imdone-pitch-feedback-welcome-386430accf01). Can comment "information for this field can be populated with the result of add_gks_vrs()", and also note this in the docstring for the function

**Comment by klaricch** [comment]
general comment, rename "gnomad_gks_batch" -> "gnomad_gks" and "gks_compute_seqloc_digest_batch" -> "gks_compute_seqloc_digest" to avoid confusion with these being associated with Hail Batch

File: gnomad/resources/grch38/gnomad.py [hunk 12]
```diff
@@ -513,9 +515,12 @@ def gnomad_gks_batch(
             " please also specify 'by_ancestry_group' to stratify by."
         )
 
-    # Call and return add_gks*() for chosen arguments.
+    # Call and return add_gks_vrs and add_gks_va for chosen arguments.
 
     # Filter to interval before adding annotations
+    ht = ht.select(ht.freq, ht.info.vrs, ht.popmax)
+    if checkpoint_path:
+        ht = ht.checkpoint(checkpoint_path, overwrite=overwrite)
```
Comment (klaricch) [inline gnomad/resources/grch38/gnomad.py:523]: rather than adding in two new params, you can change this to just use new_temp_file

**Review by klaricch — CHANGES_REQUESTED** [review]

File: gnomad/resources/grch38/gnomad.py [hunk 13]
```diff
@@ -517,10 +515,9 @@ def gnomad_gks(
 
     # Call and return add_gks_vrs and add_gks_va for chosen arguments.
 
-    # Filter to interval before adding annotations
+    # Filter to select, checkpoint, and filter before adding annotations
```
Comment (klaricch) [inline gnomad/resources/grch38/gnomad.py:518]:
```suggestion
    # Select relevant fields, checkpoint, and filter to interval before adding annotations
```

**Review by klaricch — CHANGES_REQUESTED** [review]

**Review by klaricch — APPROVED** [review]
lgtm

**Pull request merged**